sklearn/preprocessing/tests/test_polynomial.py

```diff
@@ -788,6 +793,207 @@ def test_polynomial_features_csr_X_dim_edges(deg, dim, interaction_only):
     assert_array_almost_equal(Xt_csr.A, Xt_dense)
 
 
+def test_csr_polynomial_expansion_index_overflow_non_regression():
+    """Tests to ensure that sufficiently large input configurations get
+    properly promoted to use `np.int64` for index and indptr representation
+    while preserving data integrity. Non-regression test for gh-16803.
+    """
+
+    def degree_2_calc(d, i, j, interaction_only):
+        if interaction_only:
+            return d * i - (i**2 + 3 * i) // 2 - 1 + j
+        else:
+            return d * i - (i**2 + i) // 2 + j
+
+    n_samples = 13
+    n_features = 120001
+    dtype = np.float32
+    data = np.arange(1, 5, dtype=np.int64)
+    row = np.array([n_samples - 2, n_samples - 2, n_samples - 1, n_samples - 1])
+    col = np.array([n_features - 2, n_features - 1, n_features - 2, n_features - 1])
+    X = sparse.csr_matrix(
+        (data, (row, col)),
+        shape=(n_samples, n_features),
+        dtype=dtype,
+    )
+    pf = PolynomialFeatures(interaction_only=True, include_bias=False, degree=2)
+
+    # Calculate the number of combinations a-priori, and if needed check for
+    # the correct ValueError and terminate the test early.
+    num_combinations = pf._num_combinations(
+        n_features=n_features,
+        min_degree=0,
+        max_degree=2,
+        interaction_only=pf.interaction_only,
+        include_bias=pf.include_bias,
+    )
+    if num_combinations > np.iinfo(np.intp).max:
+        msg = (
+            "The output that would result from the current configuration is too large"
+            " to be indexed"
+        )
```

Comment:
As codecov complains because our linux 32 bit CI config is not configured to collect and upload test coverage data:

https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=53305&view=logs&j=aabdcdc3-bb64-5414-b357-ed024fe8659e&t=b7b3ba55-d585-563b-a032-f235636c22b0

I think it would be great to change the CI config to do it, but maybe this is complex for some reason? Maybe @thomasjpfan remembers why we did not do it? I don't.

Reply:
Thanks @thomasjpfan.

It's weird that the codecov comments disappeared from the current PR... I swear I could see the previously and this is what prompted me to comment.

Reply:
I merged #26027 with more fixes to the codecov config. I also merged `main` into this PR to confirm this fixes the codecov reports and check all the branches are now covered.

Reply:
Ok coverage is really fixed in `main` now. I pushed another merge commit to this branch.

Reply:
This is now fixed.